apps/total-typescript/src/server/get-middleware-response.ts

```diff
@@ -31,6 +30,11 @@ export async function getMiddlewareResponse(req: NextRequest) {
     }
   }
 
+  // @ts-ignore
+  if (!req.nextauth.token && req.nextUrl.pathname.startsWith('/workshops')) {
```

Comment:
you just have to be logged in at all to access this route, no purchase checking is performed yet